mssql.go

```diff
@@ -762,7 +762,6 @@ loop:
 				//break loop
 				case []columnStruct:
 					cols = token
-					break loop
```

Comment:
>break loop [](http://example.com/codeflow?start=5&length=10)

based on the test failures in appveyor, I suspect you need to move this state management into a `Rows` method. 
`Rows` can detect if the first token after the columns is an error and do the right thing, whereas with this change you are putting the token stream in a state that `Rows` doesn't expect when there's no error.

Reply:
While moving the state management into a `Rows` method, I found that the error token is the last token, and it is not the first token after the columns.

Example
```
Query: "insert into test (f, g) output inserted.f, inserted.g values (@p1, @p2)", 1, "testValue")
Corresponding token array types: [[]mssql.columnStruct, mssql.doneInProcStruct, mssql.ReturnStatus, mssql.doneStruct]
```

It would break the loop after the `[]mssql.columnStruct`, then check if the next token (`mssql.doneInProcStruct`) is an error, but it is not, and the error is contained in the final token (mssql.doneStruct). 

I'm unsure how best to reconcile the fact that tokens after `[]mssql.columnStruct` should not be consumed because they may be intended for the user, and that the final token may contain an error which is currently not being surfaced to the user. What do you think?

Reply:
I'm attempting to sift through the tokens checking for an error, and forwarding them to a new channel/reader. I'm getting an issue with the reader blocking on scan but this may be a potential solution.

Reply:
Going through all tokens and forwarding them to a new `tokChan` does not seem to be effective because the new token channel buffer must be large enough to fit all tokens. I'm still trying to understand why `make(chan tokenStruct, 5)` is sufficient for the original token channel, but not for copying all the values to a new channel.